resource-manager/src/main/java/org/snomed/module/storage/ModuleStorageCoordinator.java

```diff
@@ -0,0 +1,602 @@
+package org.snomed.module.storage;
+
+import org.ihtsdo.otf.resourcemanager.ResourceManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.snomed.otf.script.utils.FileUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.*;
+
+/**
+ * Perform basic operations of remotely stored RF2 packages in a controlled & consistent manner.
+ */
+public class ModuleStorageCoordinator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ModuleStorageCoordinator.class);
+
+    private final ResourceManager resourceManager;
+    private final RF2Service rf2Service;
+    private final String writeDirectory;
+    private final List<String> readDirectories;
+    private final boolean allowArchive;
```

Comment:
@pgwilliams All writes to "prod" are safe because they create new data but potentially deleting existing data is dangerous, so I've added allowArchive which controls whether the archive (modify existing prod data) works. Just another step for the client/caller to be 100% sure they know what they're doing.